### ISSUE TITLE
Feature Request: Add `--check` flag to show if the tool would make changes

### ISSUE THREAD

**Issue by bradjc** [jonschlinkert/markdown-toc#86]
It is very easy to edit a markdown file and change a section heading that would change the TOC, but then forget to re-generate the TOC afterwards. Therefore, adding a check to a continuous integration service like Travis-ci to make sure that TOC are staying in sync would be useful.

I'm imagining adding a `--check` flag which would cause `markdown-toc` to return 0 if there are no changes to make, and something else if there are. But really any way to tell if `markdown-toc` should be run is sufficient.

**Comment by jonschlinkert** [comment]
Couldn't you just run `markdown-toc` if the markdown file(s) change? It would probably take the same amount of time to regenerate the toc as it would to reparse the file to check for heading changes.

**Comment by ppannuto** [comment]
I don't think this is about the time it takes, it's solving the problem where people forget to run markdown-toc. CI services shouldn't be pushing commits when they run. [rustfmt](https://github.com/rust-lang-nursery/rustfmt) as an example provides the `--diff` option, which prints the diff and returns non-zero if it would've changed anything. This is expressly for CI applications.

(I actually think --diff would be better than --check, as it shows what changed, but I could see --check being faster / easier to implement)

**Comment by jonschlinkert** [comment]
> where people forget to run markdown-toc

I don't think that's relevant, is it? It sounds like `watch` is being used. Otherwise, I'm confused at what value this would provide.

**Comment by ppannuto** [comment]
I think we're confusing things a bit here, let's give a concrete example:

I have a repository with some documentation written in markdown.

Someone comes along to improve the documentation, so they click edit in
GitHub's web view, which automatically creates a pull request.

I have Travis CI set up on my project, and I want Travis to automatically
validate that the edits made update the ToC if needed.


There's no local development environment in this case, or a place for watch
(or markdow-toc) to run. The idea of continuous integration in this case is
to make sure that the submitter of a change updated the ToC correctly,
whether by running markdown-toc or any other means.

**Comment by jonschlinkert** [comment]
> I think we're confusing things a bit here,

Yes, one of us is lol

**Comment by samestep** [comment]
I realize this is an old issue, but I wanted to add another example. Like @bradjc and @ppannuto, I use markdown-toc in CI to catch if someone edits `README.md` or `CONTRIBUTING.md` but forgets to update the table of contents. It's a polyglot project with many different linters, so earlier today I merged gradbench/gradbench#574 which consolidated them into a single command that people could use to run all of them at once. Most of [the wrappers](https://github.com/gradbench/gradbench/blob/d4fbb9dd6515e0c0f763382ee1bdce3b40a83184/crates/gradbench/src/lint.rs#L147-L271) were super simple to write, but for markdown-toc, I needed to pull in [two](https://crates.io/crates/tempfile/3.17.1) [dependencies](https://crates.io/crates/similar/2.7.0) so that I could

1. make a temporary copy of the file;
2. run markdown-toc on that;
3. check if they're equal; and
4. if not, print a diff.

None of that was necessary for any of the other tools, because having a "check" mode is standard for linters.
